tools/src/bin/rust-indexer.rs

```diff
@@ -809,3 +824,219 @@ fn main() {
+
+    for doc in &index.documents {
+        let searchfox_path = Path::new(&doc.relative_path).to_owned();
+        let searchfox_path = match scip_prefix.and_then(|prefix| scip_file.strip_prefix(prefix).ok()) {
+            Some(p) => {
+                let mut p = p.to_owned();
+                p.pop();
+                p.join(&searchfox_path)
+            },
+            None => searchfox_path,
+        };
+
+        // let local_source_path = searchfox_path_to_local_path(searchfox_path, tree_info);
+        let output_file = tree_info.out_analysis_dir.join(&searchfox_path);
+        if let Err(err) = create_output_dir(&output_file) {
+            error!(
+                "Couldn't create dir for: {}, {:?}",
+                output_file.display(),
+                err
+            );
+            continue;
+        }
+        let mut file = match File::create(&output_file) {
+            Ok(f) => f,
+            Err(err) => {
+                error!(
+                    "Couldn't open output file: {}, {:?}",
+                    output_file.display(),
+                    err
+                );
+                continue;
+            }
+        };
+        // Be chatty about the files we're outputting so that it's easier to follow
+        // the path of rust analysis generation.
+        info!(
+            "Writing analysis for '{}' to '{}'",
+            searchfox_path.display(),
+            output_file.display()
+        );
+
+        // A map from local symbol to the index in doc.symbols for this document.
+        let mut doc_symbols_to_index = HashMap::new();
+        for (index, symbol) in doc.symbols.iter().enumerate() {
+            doc_symbols_to_index.insert(symbol.symbol.clone(), index);
+        }
+
+        let lookup_symbol = |s: &str| -> Cow<SymbolInformation> {
+            match doc_symbols_to_index.get(s) {
+                Some(i) => Cow::Borrowed(&doc.symbols[*i]),
+                None => {
+                    warn!("Didn't find symbol {:?} in local symbol table", s);
+                    // Fake it till you make it? We have no info for this
+                    // symbol, so...
+                    Cow::Owned(SymbolInformation {
+                        symbol: s.to_owned(),
+                        documentation: vec![],
+                        relationships: vec![],
+                        special_fields: Default::default(),
+                    })
+                }
+            }
+        };
+
+        for occurrence in &doc.occurrences {
+            let loc = scip_range_to_searchfox_location(&occurrence.range);
+            let symbol = lookup_symbol(&occurrence.symbol);
+            {
+                let global = sanitize_symbol(&symbol.symbol);
+                let pretty = pretty_symbol(&symbol.symbol);
+
+                let source_data = WithLocation {
+                    data: AnalysisSource {
+                        source: SourceTag::Source,
+                        syntax: vec![],
```

Comment:
As per the above it turns out that our "check" infrastructure needs us to populate the syntax field.  [In the C++ indexer we explicitly put the canonical target kind in there](https://searchfox.org/mozilla-central/rev/bfa4dfd40b8fe6f1a6504e86b5ad2c5fd402bfc0/build/clang-plugin/mozsearch-plugin/MozsearchIndexer.cpp#1417-1420).

Note that if fixing this (which I think is probably a good idea), we do have production mozilla-central checks that will be impacted ([ex](https://github.com/mozsearch/mozsearch-mozilla/blob/master/mozilla-central/checks/snapshots/analysis/rust/check_glob%40build__webrender_shader__json.snap)).  The test docs at https://github.com/mozsearch/mozsearch/blob/master/docs/testing-checks.md should provide some useful context on this.

It's probably possible to fix the existing path and update the checks separate from this patch which would then not need to change the checks, but I don't have a strong opinion other than thinking we probably want the improved coverage and it's probably good for you to experience this workflow as it's one of the areas I think I've been basically doing my own thing and it would be great to be able to have others have more experience here and to be able to provide feedback/etc.

Note that the WIP test explanations do exist and capture `tracing` logging, so like right now I have at `REPO/tests/tests/checks/explanations/analysis/rust/simple.rs/simple_Loader_new__json-local.md`, the following file contents:

```json
{
  "level": "TRACE",
  "target": "tools::cmd_pipeline::builder",
  "fields": {
    "cmd": "FilterAnalysis(FilterAnalysis { file: \"simple.rs\", record_type: None, kind: Some(\"def\"), query_opts: SymbolicQueryOpts { symbol: Some(\"simple::<Loader>::new\"), identifier: None } })"
  },
  "timestamp": "2022-10-12T15:49:26.742788Z",
  "spans": [
    {
      "name": "build_pipeline",
      "arg_str": "filter-analysis simple.rs -k def -s simple::<Loader>::new\n"
    }
  ]
}
```


```json
{
  "timestamp": "2022-10-12T15:49:26.743350Z",
  "spans": [
    {
      "name": "run_pipeline_step",
      "cmd": "FilterAnalysisCommand { args: FilterAnalysis { file: \"simple.rs\", record_type: None, kind: Some(\"def\"), query_opts: SymbolicQueryOpts { symbol: Some(\"simple::<Loader>::new\"), identifier: None } } }"
    }
  ],
  "level": "TRACE",
  "fields": {
    "output_json": "{\n  \"JsonRecords\": {\n    \"by_file\": [\n      {\n        \"file\": \"simple.rs\",\n        \"records\": [\n          {\n            \"loc\": \"00037:11-14\",\n            \"target\": 1,\n            \"kind\": \"def\",\n            \"pretty\": \"simple::<Loader>::new\",\n            \"sym\": \"simple::<Loader>::new\"\n          }\n        ]\n      }\n    ]\n  }\n}"
  },
  "target": "tools::cmd_pipeline::interface"
}
```

These contents are `.gitignore`d right now because I want to improve [the template that logs the explanations](https://github.com/mozsearch/mozsearch/blob/master/tools/templates/pipeline_explainer.liquid) and also because the [tracing "cleverness" in test_check_insta](https://github.com/mozsearch/mozsearch/blob/b7c0f39354fe3ec74a74d3ad4937147ab29c213c/tools/tests/test_check_insta.rs#L129-L138) only is able to report for simple pipelines that stay on the testing thread but loses track of things when they go multi-threaded.  https://bugzilla.mozilla.org/show_bug.cgi?id=1763005 generally tracks the goal and remaining issues there.

So if you want more insight into what the checks are doing, that's a place you can find it and if you add more tracing logging, you can find that there too.  For these simple test cases that are basically just a single pipeline step, there isn't much utility there, but like for the HTML extraction it gets more interesting because you can see the intermediate values of the pipeline, for example for the HTML version of the above ([input](https://github.com/mozsearch/mozsearch/blob/master/tests/tests/checks/inputs/analysis/rust/simple.rs/simple_Loader_new__html)):

```json
{
  "spans": [
    {
      "arg_str": "filter-analysis simple.rs -k def -s simple::<Loader>::new | show-html\n",
      "name": "build_pipeline"
    }
  ],
  "target": "tools::cmd_pipeline::builder",
  "fields": {
    "cmd": "FilterAnalysis(FilterAnalysis { file: \"simple.rs\", record_type: None, kind: Some(\"def\"), query_opts: SymbolicQueryOpts { symbol: Some(\"simple::<Loader>::new\"), identifier: None } })"
  },
  "level": "TRACE",
  "timestamp": "2022-10-12T15:49:26.735229Z"
}
```


```json
{
  "target": "tools::cmd_pipeline::builder",
  "level": "TRACE",
  "timestamp": "2022-10-12T15:49:26.735335Z",
  "fields": {
    "cmd": "ShowHtml(ShowHtml)"
  },
  "spans": [
    {
      "name": "build_pipeline",
      "arg_str": "filter-analysis simple.rs -k def -s simple::<Loader>::new | show-html\n"
    }
  ]
}
```


```json
{
  "level": "TRACE",
  "fields": {
    "output_json": "{\n  \"JsonRecords\": {\n    \"by_file\": [\n      {\n        \"file\": \"simple.rs\",\n        \"records\": [\n          {\n            \"loc\": \"00037:11-14\",\n            \"target\": 1,\n            \"kind\": \"def\",\n            \"pretty\": \"simple::<Loader>::new\",\n            \"sym\": \"simple::<Loader>::new\"\n          }\n        ]\n      }\n    ]\n  }\n}"
  },
  "timestamp": "2022-10-12T15:49:26.736039Z",
  "target": "tools::cmd_pipeline::interface",
  "spans": [
    {
      "cmd": "FilterAnalysisCommand { args: FilterAnalysis { file: \"simple.rs\", record_type: None, kind: Some(\"def\"), query_opts: SymbolicQueryOpts { symbol: Some(\"simple::<Loader>::new\"), identifier: None } } }",
      "name": "run_pipeline_step"
    }
  ]
}
```


```json
{
  "target": "tools::cmd_pipeline::interface",
  "level": "TRACE",
  "fields": {
    "output_json": "{\n  \"HtmlExcerpts\": {\n    \"by_file\": [\n      {\n        \"file\": \"simple.rs\",\n        \"excerpts\": [\n          \"<div role=\\\"row\\\" id=\\\"line-37\\\" class=\\\"source-line-with-number\\\">\\n  <div role=\\\"cell\\\"><div class=\\\"cov-strip cov-no-data\\\"></div></div>\\n  <div role=\\\"cell\\\"><div class=\\\"blame-strip\\\"></div></div>\\n  <div role=\\\"cell\\\" class=\\\"line-number\\\" data-line-number=\\\"37\\\"></div>\\n  <code role=\\\"cell\\\" class=\\\"source-line\\\">    <span class=\\\"syn_reserved\\\" >pub</span> <span class=\\\"syn_reserved\\\" >fn</span> <span data-symbols=\\\"simple::<Loader>::new\\\" data-i=\\\"25\\\" >new</span>(<span data-symbols=\\\"simple::<Loader>::new::deps_dir\\\" data-i=\\\"26\\\" >deps_dir</span>: PathBuf) -> <span class=\\\"syn_reserved\\\" >Self</span> {\\n</code>\\n</div>\\n\"\n        ]\n      }\n    ]\n  }\n}"
  },
  "timestamp": "2022-10-12T15:49:26.737078Z",
  "spans": [
    {
      "name": "run_pipeline_step",
      "cmd": "ShowHtmlCommand { args: ShowHtml }"
    }
  ]
}
```